guides/typography.md

```diff
@@ -55,6 +55,18 @@ descendant native elements.
 </section>
 ```
 
+When applying the `mat-typography` CSS class, the descendent element rule will by default have
+higher specificity than the rules based on the CSS classes above.  Therefore, if you would like to
+be able to override a `mat-typography` descendent element rule using one of the CSS classes, you'll
+need to generate rules for the CSS classes as descendents so they'll have higher specificity than
+the descendent element rules like so:

```

Comment:
I'm not sure that it's very clear what the issue is from this paragraph. What do you think about something like this?

> Note that by default the `mat-typography` selectors have a very low specificity which may cause some of your own styling to conflict with them. You can work around it by increasing the selector's specificity.

Reply:
Thanks very much for the review.  So yeah, clearly it's not worded well if it's not clear what the issue is.  Sorry for that.  Maybe an example would help.

```html
<section class="mat-typography">
  <h1>This header will be styled using the default h1 from the typography configuration.  Hooray!</h1>
  <h1 class="mat-subheading-1">Unfortunately .mat-subheading-1 loses and this looks the same</h1>
</section>
```

It would be nice to be able to use .mat-typography for sensible defaults, but allow them to be overridden by the built-in classes so you can use the right semantic tags but still be able to style them according to a standard style guide.

And the reason it doesn't work is that the rules look like this:

```scss
  .mat-h1, .mat-headline, #{$selector} h1 {
    @include mat-typography-level-to-styles($config, headline);
    margin: 0 0 16px;
  }
  .mat-h4, .mat-subheading-1, #{$selector} h4 {
    @include mat-typography-level-to-styles($config, subheading-1);
    margin: 0 0 16px;
  }
```

instead of this:

```scss
  .mat-h1, .mat-headline, #{$selector} h1, #{$selector} .mat-headline {
    @include mat-typography-level-to-styles($config, headline);
    margin: 0 0 16px;
  }
  .mat-h4, .mat-subheading-1, #{$selector} h4, #{$selector} .mat-subheading-1 {
    @include mat-typography-level-to-styles($config, subheading-1);
    margin: 0 0 16px;
  }
```

I suggested this in an issue but was told that it's by design.  So I thought it would be good to let people know how they can still achieve this setup in the guide.

Sorry for the lonnng explanation!